nasa/ow_simulator
# Oceanwater 701 camera power consumption

## Linked Issues:
| EPIC ⚡ | [OCEANWATER-697](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-697) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-701](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-701) |

## Blocker PRs
https://github.com/nasa/ow_simulator/pull/384 Needs to be reviewed and merged before reviewing of this PR can take place. 

## Summary of Changes
* Camera controller, an always on device, now consumes power similar to lander lights and mechanical power. It is settable in `system.cfg` as `power_camera_controller`.
* Each camera being exposed draws a power of 2.0 W (see Caveats).

## Test
1. In system.cfg: change both print_debug and inputs_print_debug to true.
2. Start any simulation world.
3. You should see the output 
```
[ INFO:/power_system_node] M00 INPUT power: 5.100000.  volts: 17.640419.  tmp: 21.260296
```
4. Close the simulation
5. Change the value of `power_camera_controller` in system.cfg to `6.0`
6. Run the simulation, and you should see the following output
```
[ INFO:/power_system_node] M00 INPUT power: 6.350023.  volts: 17.699259.  tmp: 18.253528
```
Indicating that the extra camera controller power load of 5.0 has been distributed across all 4 power cells. 

## Caveat
The power draw from cameras being exposed has been implemented and is settable as `power_camera_exposed_coefficient` in system.cfg; however, it appears our camera model does not actually take any simulation time to expose, no matter how high you set the exposure time. Meaning the value does not work as expected. The implementation is left for in case this camera exposure bug is addressed in the future. See [OW-1239](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1239)
